### PR TITLE
Post alpha refactor

### PR DESCRIPTION
**Pull request by hkalodner**
This pull request includes a large amount of cleanup and refactoring, but shouldn't change any functionality.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,8 +35,14 @@ require (
...
+	golang.org/x/net v0.0.0-20190613194153-d28f0bde5980 // indirect
+	golang.org/x/sys v0.0.0-20190614160838-b47fdc937951 // indirect
+	golang.org/x/text v0.3.2 // indirect
+	golang.org/x/tools v0.0.0-20190614205625-5aca471b1d59 // indirect
 	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
 	gopkg.in/natefinch/npipe.v2 v2.0.0-20160621034901-c1b8fa8bdcce // indirect
 	gopkg.in/urfave/cli.v1 v1.20.0 // indirect
 )
+
+replace github.com/offchainlabs/arb-avm => ../arb-avm
```
Comment (ughe) [inline go.mod]: Is this re-write rule still supposed to be in this commit?

File: checkpoint/checkpoint.go [hunk 2]
```diff
@@ -688,12 +691,17 @@ func (cp *Checkpointer) restoreMachine_inTxn(txn *badger.Txn, keySuffix []byte)
...
 		return nil, err
 	}
 
-	return vm.RestoreMachine(codeOps, vals[0], vals[1], vals[2], vals[3], vals[4], vals[5], sizeLimit), nil
+	errHandler, ok := vals[5].(value.CodePointValue)
+	if !ok {
+		return nil, errors.New("6th value must be a codepoint")
+	}
+
+	return vm.RestoreMachine(codeOps, vals[0], vals[1], vals[2], vals[3], vals[4], errHandler, sizeLimit), nil
```
Comment (ughe) [inline checkpoint/checkpoint.go]: This line does not compile: `cannot use vals[0] (type value.Value) as type value.CodePointValue in argument to vm.RestoreMachine: need type assertion` for `vals[0]` through `vals[4]`

File: challenge/challenger/challenger.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		return waitingContinuing{
+			bot.Config,
+			bot.preconditions[ev.ChallengedAssertion],
+			bot.assertions[ev.ChallengedAssertion],
+			bot.challengedInbox,
+			bot.challengedState,
+			deadline,
+		}, nil
+	default:
+		return nil, &challenge.Error{nil, "ERROR: continuing: VM state got unsynchronized"}
```
Comment (ughe) [inline challenge/challenger/challenger.go]: `go vet ./...` points out unkeyed fields here a couple places as well as in `observer.go` and `defender.go`. Should these warning be ignored?

File: ethvalidator/follower.go [hunk 4]
```diff
@@ -99,267 +97,221 @@ func NewValidatorFollower(
...
+			case *valmessage.ValidatorRequest_Unanimous:
+				err := m.HandleUnanimousRequest(request.Unanimous, value.NewHashFromBuf(req.RequestId))
+				if err != nil {
+					log.Printf("Follower error while trying to handle unanimous assertion request from coordinator")
+				}
+			case *valmessage.ValidatorRequest_UnanimousNotification:
+				requestInfo := m.unanimousRequests[value.NewHashFromBuf(req.RequestId)]
+				if request.UnanimousNotification.Accepted {
+					sigs := make([][]byte, len(request.UnanimousNotification.Signatures))
+					for _, sig := range request.UnanimousNotification.Signatures {
```
Comment (ughe) [inline ethvalidator/follower.go]: Running staticcheck suggests `Use a single append to concatenate two slices`:
```should replace loop with sigs = append(sigs, request.UnanimousNotification.Signatures...) (S1011)```

File: coordinator/server.go [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+		case evm.Revert:
+			reply.ReturnVal = hexutil.Encode(result.ReturnVal)
+			reply.Success = false
+		}
+		return nil
+	case err := <-errChan:
+		fmt.Println("Call failed")
+		return err
+	}
+}
```
Comment (ughe) [inline coordinator/server.go]: There should be a newline at the end of this file (not too important)

File: valmessage/requests.go [hunk 6]
```diff
@@ -99,54 +54,7 @@ type UnanimousUpdateResults struct {
...
-		r.ResultChan <- false
-	}()
-}
-
-func (r DisputableAssertionRequest) NotifyAccepted() {
-	go func() {
-		r.ResultChan <- true
-	}()
-}
+}
```
Comment (ughe) [inline valmessage/requests.go]: Another missing newline

File: go.mod (hunk 1 above)
Comment (hkalodner) [reply go.mod]: Ah whoops. Definitely not supposed to be in there. Good catch

Comment (hkalodner) [reply go.mod]: Added new commit to remove this.

File: checkpoint/checkpoint.go (hunk 2 above)
Comment (hkalodner) [reply checkpoint/checkpoint.go]: https://github.com/OffchainLabs/arb-avm/pull/12 submitted to fix this bug

File: challenge/challenger/challenger.go (hunk 3 above)
Comment (hkalodner) [reply challenge/challenger/challenger.go]: All unkeyed fields now fixed

File: ethvalidator/follower.go (hunk 4 above)
Comment (hkalodner) [reply ethvalidator/follower.go]: Just pushed a fix for that.

File: coordinator/server.go (hunk 5 above)
Comment (hkalodner) [reply coordinator/server.go]: Fixed for this file and a few others.

File: valmessage/requests.go (hunk 6 above)
Comment (hkalodner) [reply valmessage/requests.go]: Fixed

**Pull request merged**